src/components/Button/Button.vue

```diff
@@ -0,0 +1,305 @@
+ -
+ -->
+
+<docs>
+
+### General description
+```
+<Button
+	text="Example text" />
+<Button
+	text="Example text"
+	color="primary" />
+<Button
+	text="Example text"
+	color="success" />
+<Button
+	text="Example text"
+	color="error" />
+<Button
+	text="Example text"
+	color="warning" />
+```
+### Usage
+
+</docs>
+
+<template>
+	<button
+		v-ripple
+		class="button-vue"
+		v-bind="$attrs"
+		:class="buttonClassObject"
+		:aria-label="ariaLabel"
+		@click="handleClick"
+		@blur="handleBlur"
+		@keyup.tab.exact="handleTabUp">
+		<span class="button-vue__wrapper">
+			<span v-if="hasIcon" class="button-vue__icon">
+				<slot />
+			</span>
+			<span v-if="hasText" class="button-vue__text">{{ text }}</span>
+		</span>
+	</button>
+</template>
+<script>
+import Ripple from '../../directives/Ripple'
+
+export default {
+	name: 'Button',
+
+	directives: {
+		ripple: Ripple,
+	},
+
+	props: {
+		/**
+		 * The text of the button
+		 */
+		text: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * Specifies the button color
+		 * Accepted values: primary, error, warning, success. If left empty,
+		 * the default button style will be used.
+		 */
+		color: {
+			type: String,
+			validator(value) {
+				return ['primary', 'error', 'warning', 'success', ''].indexOf(value) !== -1
+			},
+			default: '',
+		},
+
+		/**
+		 * Specifies whether the button should span all the available width.
+		 * By default, buttons span the whole width of the container.
+		 */
+		wide: {
+			type: Boolean,
+			default: true,
+		},
+
+		/**
+		 * Always provide an aria-label to your button and try to make it more
+		 * specific than the button's title. E.g. if the title of the button is
+		 * "send" in the Mail app, the label could be "Send email"
+		 */
+		ariaLabel: {
+			type: String,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			/**
+			 * Keeps track of whether the element's focus status is due to having
+			 * tabbed to it. We use this to display a thick 'focus outline' only
+			 * when the user is navigating with the keyboard.
+			 */
+			tabbed: false,
+		}
+	},
+
+	computed: {
+		hasText() {
+			return this.text
+		},
+
+		hasIcon() {
+			return this.$slots.default !== undefined
+		},
+
+		iconOnly() {
+			return this.hasIcon && !this.hasText
+		},
+
+		textOnly() {
+			return !this.hasIcon && this.hasText
+		},
+
+		iconAndText() {
+			return this.hasIcon && this.hasText
+		},
+
+		// Classes applied to the button element
+		buttonClassObject() {
+			return {
+				// If icon only, some additional css rules are required
+				'button-vue--icon-only': this.iconOnly,
+				'button-vue--text-only': this.textOnly,
+				'button-vue--icon-and-text': this.iconAndText,
+				[`button-vue--vue-${this.color}`]: this.color,
+				'button-vue--wide': this.wide,
+				'button-vue--tabbed': this.tabbed,
+			}
+		},
+	},
+
+	methods: {
+		// Forward the click event to the parent component
+		handleClick(event) {
+			this.$emit('click', event)
+		},
```

Comment:
Why is this needed?

Reply:
You can also bind all listeners on the button if that is your issue :wink:

Reply:
It's not is it? It could be just instance of the <Button /> component?
like:
```
<Button @click="...">
```

Reply:
Any drawbacks to that? ^

Reply:
Well, theoritecally you should be able to listen to the click event without having the click listener within the component.
You can use `@click.native` if needed for example. But I'm wondering why `@click` would not just work :thinking:

Reply:
I'll test this but I think it will work. So I'll likely remove this too!